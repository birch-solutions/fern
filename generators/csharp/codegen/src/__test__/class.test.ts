import { Access, CSharp } from "..";

describe("class", () => {
    it("basic", () => {
        const plant = new CSharp.Class({
            name: "Plant",
            namespace: "Fern",
            partial: true,
            access: Access.Public
        });
        plant.addField(
            new CSharp.Field({
                name: "watered",
                type: CSharp.Type.boolean(),
                access: Access.Public,
                get: true,
                init: true,
                annotations: [
                    new CSharp.Annotation({
                        reference: new CSharp.ClassReference({
                            name: "JsonProperty",
                            namespace: "System.Text.Json.Serialization"
                        }),
                        argument: '"watered"'
                    })
                ]
            })
        );
        expect(plant.name).toBe("Plant");
        expect(plant.toString()).toMatchSnapshot();
    });
});
