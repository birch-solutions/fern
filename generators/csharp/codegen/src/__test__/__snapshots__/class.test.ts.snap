// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class basic 1`] = `
"
namespace Fern

public partial class Plant
{
    [JsonPropertyJsonProperty("watered")]
    bool public undefined watered { get; init; } 
}
"
`;
